pkg/search/postgres/joins_test.go

```diff
@@ -0,0 +1,613 @@
+//go:build sql_integration
+// +build sql_integration
+
+package postgres
+
+import (
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/postgres/walker"
+	"github.com/stretchr/testify/assert"
+)
+
+type testSet struct {
+	Root            *joinTreeNode
+	ReachableFields map[string]searchFieldMetadata
+	ExpectedRoot    *joinTreeNode
+}
+
+func getTestData() map[string]testSet {
+	data := map[string]testSet{
+		"nil join tree": {
+			Root:         nil,
+			ExpectedRoot: nil,
+		},
+		"single table": {
+			Root: &joinTreeNode{
+				currNode: &walker.Schema{
```

Comment:
How about create helper function to create objects so this will be easier to read?
```go
func t(id string) *walker.Schema {
	return &walker.Schema{
		Table: id,
	}
}
```

Reply:
```go
func node(tid string) *joinTreeNode {
	return &joinTreeNode{
		currNode: t(tid),
	}
}

func t(id string) *walker.Schema {
	return &walker.Schema{
		Table: "t" + id,
	}
}

func pairs(this, other string) []walker.ColumnNamePair {
	return []walker.ColumnNamePair{{
		ColumnNameInThisSchema:  this,
		ColumnNameInOtherSchema: other,
	}}
}

func pair(this, other string) walker.ColumnNamePair {
	return walker.ColumnNamePair{
		ColumnNameInThisSchema:  this,
		ColumnNameInOtherSchema: other,
	}
}
```
If you use above functions then code will be much easier to read
e.g.
```go
			ExpectedRoot: &joinTreeNode{
				currNode: t("1"),
				children: map[*joinTreeNode][]walker.ColumnNamePair{
					{
						currNode: t("2"),
						children: map[*joinTreeNode][]walker.ColumnNamePair{
							node("3"): pairs("t2_c1", "t3_c1"),
						},
					}: pairs("t1_c1", "t2_c1"),
				},
			},
```